spec/javascripts/govuk_publishing_components/load-analytics.spec.js

```diff
@@ -0,0 +1,323 @@
+      expect(window.GOVUK.analyticsGa4.vars).toEqual(expected)
+    })
+
+    it('sets the correct staging variables', function () {
+      expected.environment = 'staging'
+      expected.auth = 'oJWs562CxSIjZKn_GlB5Bw'
+      expected.preview = 'env-5'
+
+      window.GOVUK.loadAnalytics.loadGa4('www.staging.publishing.service.gov.uk')
+      delete window.GOVUK.analyticsGa4.vars.internalDomains
+      expect(window.GOVUK.analyticsGa4.vars).toEqual(expected)
+    })
+
+    it('sets the correct integration variables', function () {
+      expected.environment = 'integration'
+      expected.auth = 'C7iYdcsOlYgGmiUJjZKrHQ'
+      expected.preview = 'env-4'
+      window.GOVUK.loadAnalytics.loadGa4('www.integration.publishing.service.gov.uk')
+      delete window.GOVUK.analyticsGa4.vars.internalDomains
+      expect(window.GOVUK.analyticsGa4.vars).toEqual(expected)
+    })
+
+    it('sets the correct development variables', function () {
+      expected.environment = 'development'
+      expected.auth = 'bRiZ-jiEHtw6hHpGd6dF9w'
+      expected.preview = 'env-3'
+      window.GOVUK.loadAnalytics.loadGa4('localhost')
+      delete window.GOVUK.analyticsGa4.vars.internalDomains
+      expect(window.GOVUK.analyticsGa4.vars).toEqual(expected)
+    })
+
+    it('doesnt load GA4 variables if initialiseGA4 is set to false', function () {
+      window.GOVUK.loadAnalytics.ga4EnvironmentVariables.production.initialiseGA4 = false
+      window.GOVUK.loadAnalytics.loadGa4('www.gov.uk')
+      expect(window.GOVUK.analyticsGa4.vars).toEqual(null)
+      window.GOVUK.loadAnalytics.ga4EnvironmentVariables.production.initialiseGA4 = true
+    })
+  })
+
+  describe('for UA', function () {
+    beforeEach(function () {
+      window.GOVUK.analyticsVars = null
+      expected = {
+        gaPropertyCrossDomain: 'UA-145652997-1',
+        primaryLinkedDomains: ['account.gov.uk'],
+        linkedDomains: linkedDomains
+      }
+    })
+
+    it('loads the correct UA values on production', function () {
+      expected.gaProperty = 'UA-26179049-1'
+      window.GOVUK.loadAnalytics.loadUa('www.gov.uk')
+      expect(window.GOVUK.analyticsVars).toEqual(expected)
+    })
+
+    it('loads the correct UA values on staging', function () {
+      expected.gaProperty = 'UA-26179049-20'
+      window.GOVUK.loadAnalytics.loadUa('www.staging.publishing.service.gov.uk')
+      expect(window.GOVUK.analyticsVars).toEqual(expected)
+    })
+
+    it('loads the correct UA values on integration', function () {
+      expected.gaProperty = 'UA-26179049-22'
+      window.GOVUK.loadAnalytics.loadUa('www.integration.publishing.service.gov.uk')
+      expect(window.GOVUK.analyticsVars).toEqual(expected)
+    })
+
+    it('loads the correct UA values on development', function () {
+      expected.gaProperty = 'UA-UNSET'
+      expected.gaPropertyCrossDomain = 'UA-UNSET'
+      window.GOVUK.loadAnalytics.loadUa('localhost')
+      expect(window.GOVUK.analyticsVars).toEqual(expected)
```

Comment:
This test has highlighted to me that the linkedDomains appear to be loaded on development and unrecognised domains - should we be preventing that?

Reply:
@andysellick I believe it does that in our current implementation, which is why I kept it like that.

Reply:
@andysellick I believe `linkedDomains` is loaded on development/unrecognised domains in our current environment, which is why I've kept it the same, as I don't want to affect how universal analytics works.

Reply:
Okay, thanks.